### DISCUSSION TITLE
Filaments routes are not registered when using Filament::navigation in a serviceprovider

### DISCUSSION THREAD

**Discussion by ehs4nm** [filamentphp/filament#1853] [General]
Hi
Creating a new provider and registered it as the last service. In boot/register method of the provider added:
```
        Filament::registerUserMenuItems([
            UserMenuItem::make()
                ->label('Orders')
                ->url(route('filament.resources.orders.index'))
                ->icon('heroicon-s-cog'),
            // ...
        ]);
```

In tinker or anywhere else the route works. but when the app load filaments routes hasn't registered yet and shows error:
Route [filament.resources.orders.index] not defined

**Comment by danharrin** [comment]
Wrap it in `Filament::serving()`:

```php
Filament::serving(function () {
    Filament::registerUserMenuItems([
        UserMenuItem::make()
            ->label('Orders')
            ->url(route('filament.resources.orders.index'))
            ->icon('heroicon-s-cog'),
        // ...
    ]);
});
```

This means that the code will only run on Filament requests, when everything has properly loaded.

**Reply by ehs4nm** [reply]
It's great. It would be nice to be added to doc though:
https://filamentphp.com/docs/2.x/admin/navigation#registering-custom-navigation-items

Thanks.
